pysb/bng.py

```diff
@@ -103,12 +104,16 @@ class BngBaseInterface(object):
     @abc.abstractmethod
     def __init__(self, model, verbose=False, cleanup=False,
                  output_prefix=None, output_dir=None):
+        self._base_file_stem = 'pysb'
         self.verbose = verbose
         self.cleanup = cleanup
         self.output_prefix = 'tmpBNG' if output_prefix is None else \
             output_prefix
-        self.generator = BngGenerator(model)
-        self._check_model()
+        if model:
```

Comment:
Oh I see, I hadn't realized you could do that. Perhaps the signature here should have `model=None` then to be more explicit that it's optional. Then in any method where a model is needed, the first thing it does is raise an exception if self.model is None. (Also, why not keep a reference to model in self rather than the property that fetches it from the generator?)